install_config/configure_kerberos_for_sso.adoc

```diff
@@ -0,0 +1,174 @@
+usage property set to `TLS Web client authentication`.
+[IMPORTANT]
+====
+This certificate is critical as it will be fully trusted by the cluster to
+provide user identity data. A compromise of this certificate will allow an
+attacker to impersonate any user in the cluster.
+====
+In our example we create a certificate with the subject: `CN=oauth-proxy-client`
+
+===== Configuring Apache
+
+Two modules are required for a successful configuration: mod_request and
+mod_auth_gssapi, make sure the following configuration lines are uncommented
+in the proper configurtion file (usually something like :
+
+----
+LoadModule request_module modules/mod_request.so
+LoadModule auth_gssapi_module modules/mod_auth_gssapi.so
+----
+
+The next step is to add virtual host directives to allow Apache to respond to
+queries made to the DNS name we chose earlier:
+
+----
+# Nothing needs to be served over HTTP.  This virtual host simply redirects to
+# HTTPS.
+<VirtualHost *:80>
+  DocumentRoot /var/www/html
+  RewriteEngine              On
+  RewriteRule     ^(.*)$     https://%{HTTP_HOST}$1 [R,L]
+</VirtualHost>
+
+<VirtualHost *:443>
+  # ServerName needs to match the DNS name we chose and the SAN of the serving
+  # certificate that was generated as well as the hostname of the Service
+  # Principal Name created to obtain the keys in the keytab.
+  ServerName auth-proxy.krb.example.com
+
+  DocumentRoot /var/www/html
+  SSLEngine On
+  SSLCertificateFile /etc/pki/tls/certs/auth-proxy.krb.example.com.crt
+  SSLCertificateKeyFile /etc/pki/tls/private/auth-proxy.krb.example.com.key
+  SSLCACertificateFile /etc/pki/CA/certs/ca.crt
+
+  SSLProxyEngine On
+  SSLProxyCACertificateFile /etc/pki/CA/certs/ca.crt
+  # It's critical to enforce client certificates on the Master.  Otherwise
+  # requests could spoof the X-Remote-User header by accessing the Master's
+  # /oauth/authorize endpoint directly.
+  SSLProxyMachineCertificateFile /etc/pki/tls/certs/auth-proxy-client.crt
+
+  # Send all requests to the console
+  RewriteEngine On
+  RewriteRule   ^/console(.*)$   https://%{HTTP_HOST}:8443/console$1 [R,L]
+
+  # In order to using the challenging-proxy an X-Csrf-Token must be present.
+  RewriteCond %{REQUEST_URI} ^/challenging-proxy/?
+  RewriteCond %{HTTP:X-Csrf-Token} ^$ [NC]
+  RewriteRule ^.* - [F,L]
+
+  <Location /challenging-proxy/oauth/authorize>
+    # NOTE: this is the name of the cluster API Servers
+    ProxyPass https://cluster.krb.example.com:8443/oauth/authorize
+    AuthName "SSO Login"
+    AuthType GSSAPI
+    Require valid-user
+    RequestHeader set X-Remote-User %{REMOTE_USER}s
+
+    GssapiCredStore keytab:/etc/httpd/protected/auth-proxy.keytab
+    # Enable the following if you want to allow users to fallback
+    # to password based authntication when they do not have a client
+    # configured to perform kerberos authentication
+    GssapiBAsicAuth On
+  </Location>
+
+  <Location /login-proxy/oauth/authorize>
```

Comment:
Are these stanzas in the same file? Or different files?

Reply:
same file and same section (within virtualhost definition)